### PR TITLE
refactor: [M3-7859]: Upgrade MSW to 2.2.3

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This update brings MSW 2.2.3 to Cloud Manager

It is a [major update with breaking changes](https://mswjs.io/docs/migrations/1.x-to-2.x/), hence the large amount of files touched.

## Changes  🔄
- Update MSW version
- Regenerate file in public dir (needed to bump `yargs-parser` for that purpose)
- Update all handlers in `serverHandlers.ts` (it was not fun)
- Update all tests

## Preview 📷
No visual change to be expected

### Prerequisites
- Get yourself familiar with the new API and breaking changes: https://mswjs.io/docs/migrations/1.x-to-2.x/
- checkout PR
- run `yarn clean`
- run `yarn && yarn up`

### Verification steps
- turn on MSW and confirm no regression with mock data
- run test suite and confirm all pass

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/public/mockServiceWorker.js
Comment (abailly-akamai) [inline packages/manager/public/mockServiceWorker.js:1]: this is all generated via `npx msw init`

File: packages/manager/public/mockServiceWorker.js [hunk 1]
```diff
@@ -86,17 +87,6 @@ self.addEventListener('message', async function (event) {
 
 self.addEventListener('fetch', function (event) {
   const { request } = event
-  const accept = request.headers.get('accept') || ''
-
-  // Bypass launch darkly
-  if (request.url.includes("launchdarkly.com")) {
-    return
-  }
```
Comment (bnussman-akamai) [inline packages/manager/public/mockServiceWorker.js:94]: Can you manually add back these 4 lines? 🥺 

I know it's stupid, but Launch Darkly won't work in Safari with the MSW on without this code 😖

File: packages/manager/src/mocks/serverHandlers.ts [hunk 2]
```diff
@@ -1663,74 +1628,77 @@ export const handlers = [
...
           eventWithSpecialCharacters,
         ])
-      )
-    );
-  }),
-  rest.get('*/support/tickets', (req, res, ctx) => {
+      );
+    },
+    {
+      once: true,
```
Comment (bnussman-akamai) [inline packages/manager/src/mocks/serverHandlers.ts:1634]: My dev tools are showing a lot of `/v4/account/events` failures and I'm thinking we could fix that by mocking an empty paginated response after this initial response.

Screenshot 2024-03-14 at 3 45 22 PM

**Review by bnussman-akamai — COMMENTED** [review]
This is great! Thank you for taking this on 👏 🎉  

I left two small requests

File: packages/manager/public/mockServiceWorker.js (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/public/mockServiceWorker.js]: sure, not a big deal - done!

File: packages/manager/src/mocks/serverHandlers.ts (hunk 2 above)
Comment (abailly-akamai) [reply packages/manager/src/mocks/serverHandlers.ts:1629]: hmm i am not seeing this - the once seems to work fine for me in chrome/safari

**Review by bnussman-akamai — APPROVED** [review]
🎉🙏🏼

**Review by mjac0bs — APPROVED** [review]
🚀  Thank you for doing this; I know it was tedious! 

Tests passing and the CRUD operations I verified with the MSW mock data looked good. 
Screenshot 2024-03-18 at 10 51 07 AM

**Comment by jdamore-linode** [comment]
> (Picture of passing tests)

Surprisingly I'm getting a lot of failures! 

Screenshot 2024-03-18 at 2 18 02 PM

The fact that it's passing in CI and on other machines makes me think it's probably a performance issue with my Mac, but @abailly-akamai would you mind if I take a closer look at this / try reproducing / examining any notices and warnings just to rule out any issues before merging? I'll try to follow up before leaving today!

**Review by jdamore-linode — APPROVED** [review]

**Pull request merged**